### PR TITLE
docs: added docs for drift

### PR DESCRIPTION
**Pull request by njtran**
Fixes # 

**Description**
Adds docs on how to enable drift detection for drift automatic deprovisioning.
**How was this change tested?**

* make website

**Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/concepts/deprovisioning.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ There are both automated and manual ways of deprovisioning nodes provisioned by
...
 * **Consolidation**: Karpenter works to actively reduce cluster cost by identifying when nodes can be removed as their workloads will run on other nodes in the cluster and when nodes can be replaced with cheaper variants due to a change in the workloads.
 * **Interruption**: If enabled, Karpenter will watch for upcoming involuntary interruption events that could affect your nodes (health events, spot interruption, etc.) and will cordon, drain, and terminate the node(s) ahead of the event to reduce workload disruption.
+* **Drift**: If enabled, Karpenter will deprovision nodes that have drifted from provisioning specifications. Once the node is annotated as drifted, Karpenter will deprovision the nodes and provision replacement nodes with the correct provisioning requirements when needed.
```
Comment (bwagner5) [inline website/content/en/preview/concepts/deprovisioning.md:27]: This seems like we support more than AMIs right now, can you clarify like you do below?

**Review by bwagner5 — CHANGES_REQUESTED** [review]

**Review by bwagner5 — APPROVED** [review]
LGTM

File: website/content/en/preview/concepts/deprovisioning.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ There are both automated and manual ways of deprovisioning nodes provisioned by
...
 * **Consolidation**: Karpenter works to actively reduce cluster cost by identifying when nodes can be removed as their workloads will run on other nodes in the cluster and when nodes can be replaced with cheaper variants due to a change in the workloads.
 * **Interruption**: If enabled, Karpenter will watch for upcoming involuntary interruption events that could affect your nodes (health events, spot interruption, etc.) and will cordon, drain, and terminate the node(s) ahead of the event to reduce workload disruption.
+* **Drift**: If enabled, Karpenter will deprovision nodes that have drifted from provisioning specifications. Once the node is annotated as drifted, Karpenter will deprovision the nodes and provision replacement nodes with the correct provisioning requirements when needed. Currently, Karpenter will only automatically mark nodes as drifted in the case of a drifted AMI.
```
Comment (ellistarn) [inline website/content/en/preview/concepts/deprovisioning.md:27]:
```suggestion
* **Drift**: Karpenter will deprovision nodes that have drifted from provisioning specifications. Once the node is annotated as drifted, Karpenter will deprovision the nodes and provision replacement nodes with the correct provisioning requirements when needed. Currently, Karpenter will only automatically mark nodes as drifted in the case of a drifted AMI.
```

File: website/content/en/preview/concepts/deprovisioning.md [hunk 3]
```diff
@@ -128,6 +129,29 @@ data:
   ...
 ```
 
+## Drift
```
Comment (ellistarn) [inline website/content/en/preview/concepts/deprovisioning.md:132]: I think we need some language around how drift is handled with respect to other terminations (e.g. consolidation, etc)

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**